R/separate_longer.R

```diff
@@ -0,0 +1,255 @@
+#'
+#' @param data A Polars DataFrame or LazyFrame.
+#' @param cols <[`tidy-select`][tidyr_tidy_select]> Column(s) to separate.
+#' @param delim The delimiter string to split on. For `separate_longer_delim_polars()` only.
+#' @param width The width of each piece. For `separate_longer_position_polars()` only.
+#' @param ... These dots are for future extensions and must be empty.
+#' @param keep_empty If `TRUE`, empty strings are kept in the output.
+#'   If `FALSE` (the default), empty strings are dropped.
+#'   `NA` values are always kept.
+#'
+#' @return A Polars DataFrame or LazyFrame with the specified column(s) split
+#'   into rows.
+#'
+#' @seealso [tidyr::separate_longer_delim()], [tidyr::separate_longer_position()]
+#'
+#' @name separate_longer
+#' @export
+#' @examplesIf require("dplyr", quietly = TRUE) && require("tidyr", quietly = TRUE)
+#' library(polars)
+#' library(tidypolars)
+#'
+#' # separate_longer_delim_polars: split by delimiter
+#' df <- pl$DataFrame(
+#'   id = 1:3,
+#'   x = c("a,b,c", "d,e", "f")
+#' )
+#' separate_longer_delim_polars(df, x, delim = ",")
+#'
+#' # Multiple columns with broadcasting, the same as tidyr behavior
+#' df2 <- pl$DataFrame(
+#'   id = 1:2,
+#'   x = c("a,b", "c,d"),
+#'   y = c("1,2", "3,4")
+#' )
+#' separate_longer_delim_polars(df2, c(x, y), delim = ",")
+#'
+#' # Multiple columns with broadcasting
+#' df3 <- pl$DataFrame(
+#'   id = 1:5,
+#'   x = c("a,b", NA, "", "c", ""),
+#'   y = c("1", "2,3", "4,5", NA, "")
+#' )
+#' separate_longer_delim_polars(df3, c(x, y), delim = ",")
+#'
+#' # separate_longer_position_polars: split by fixed width
+#' df4 <- pl$DataFrame(
+#'   id = 1:3,
+#'   x = c("abcd", "efg", "hi")
+#' )
+#' separate_longer_position_polars(df4, x, width = 2)
+#'
+#' # keep_empty example: control whether empty strings are preserved
+#' df5 <- pl$DataFrame(
+#'   id = 1:4,
+#'   x = c("ab", "", "ef", NA)
+#' )
+#' separate_longer_position_polars(df5, x, width = 2)
+#' separate_longer_position_polars(df5, x, width = 2, keep_empty = TRUE)
+#'
+#' # Multiple columns with broadcasting
+#' df6 <- pl$DataFrame(
+#'   id = 1:3,
+#'   x = c("a", "bc", "def"),
+#'   y = c("12", "345", "67")
+#' )
+#' # Shorter strings are recycled to match the longest in each row
+#' separate_longer_position_polars(df6, c(x, y), width = 2)
+separate_longer_delim_polars <- function(
+  data,
+  cols,
+  delim,
+  ...
+) {
+  check_polars_data(data)
+  rlang::check_dots_empty()
+  rlang::check_required(cols)
+  check_string(delim)
+
+  # Capture the column expression and filter to only string columns
+  col_names <- tidyselect_named_arg(data, rlang::enquo(cols)) |>
+    intersect(names(data$select(pl$col(pl$String))))
+
+  # If no string columns, return data unchanged (like tidyr)
+  if (length(col_names) == 0) {
+    return(add_tidypolars_class(data))
+  }
+
+  # Split each column by delimiter and convert to list
+  out <- data$with_columns(
+    do.call(pl$col, as.list(col_names))$str$split(delim)
```

Comment:
```suggestion
    pl$col(!!!col_names)$str$split(delim)
```

Also needed in `separate_longer_position_polars()`